src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5222,9 +5223,12 @@ public static IEnumerable<CompletionResult> CompleteVariable(string variableName
 
         private static readonly string[] s_variableScopes = new string[] { "Global:", "Local:", "Script:", "Private:" };
 
-        private static readonly char[] s_charactersRequiringQuotes = new char[] {
+        private static readonly SearchValues<char> s_charactersRequiringQuotes = SearchValues.Create(new char[] {
             '-', '`', '&', '@', '\'', '"', '#', '{', '}', '(', ')', '$', ',', ';', '|', '<', '>', ' ', '.', '\\', '/', '\t', '^',
-        };
+        });
+
+        private static bool ContainsCharactersRequiringQuotes(ReadOnlySpan<char> text)
+            => text.ContainsAny(s_charactersRequiringQuotes);
```

Comment:
Can we use the method https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs,532 ?